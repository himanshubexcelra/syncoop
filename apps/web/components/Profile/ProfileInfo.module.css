.main {
    padding: 20px;
}

.box {
    height: 230px;
    background: var(--white);
    box-shadow: 0px 8px 12px 0px var(--tabBoxShadow), 0px 0px 4px 0px var(--tabBoxShadow);
    padding-left: 21px;
    padding-right: 30px;
    padding-top: 19px;
}

.type {
    color: var(--themeGreyColor);
}

.output {
    color: var(--black);
}

.grid {
    display: grid;
    grid-template-columns: 30% 70%;
}