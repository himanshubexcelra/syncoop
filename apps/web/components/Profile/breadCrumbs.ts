import { BreadCrumbsObj } from "@/lib/definition";

export const getProfileBreadCrumbs = (
    isMyProfile: boolean,
    myRoles: string[],
    orgDetailLoggedIn: { name: string } | null
): BreadCrumbsObj[] => {
    return [
        {
            label: 'Home',
            svgPath: '/icons/home-icon.svg',
            svgWidth: 16,
            svgHeight: 16,
            href: '/'
        },
        ...(isMyProfile
            ? [
                {
                    label: 'Profile',
                    svgPath: '/icons/profile-icon-sm-active.svg',
                    svgWidth: 16,
                    svgHeight: 16,
                    href: '/profile',
                    isActive: true
                }
            ]
            : [
                {
                    label: `${['admin', 'org_admin'].some((role) => myRoles?.includes(role)) ? 'Admin: ' : ''}${orgDetailLoggedIn?.name ?? ''}`,
                    svgPath: "/icons/admin-inactive-icon.svg",
                    svgWidth: 16,
                    svgHeight: 16,
                    href: "/dashboard",
                },
                {
                    label: 'Profile',
                    svgPath: '/icons/profile-icon-sm-active.svg',
                    svgWidth: 16,
                    svgHeight: 16,
                    href: '/profile',
                    isActive: true
                }
            ]
        )
    ];
};