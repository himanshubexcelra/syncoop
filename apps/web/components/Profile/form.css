.create-popup {
    .dx-popup-normal {
        background: var(--themelightGreyColor);
        box-shadow: -18px 0px 18px 0px rgba(0, 0, 0, 0.25);
    }

    .dx-popup-title {
        border-bottom: 0;
    }

    .dx-field-item:not(.dx-field-item-has-group):not(.dx-field-item-has-tabs):not(.dx-first-row):not(.dx-label-v-align) {
        padding-top: 10px;
    }

    .dx-toolbar-item-content {
        color: var(--themeDarkBlueColor);
        text-align: center;
        font-family: Lato;
        font-size: 28px;
        font-style: normal;
        font-weight: 700;
        line-height: 42px;
    }
}

.dx-form {
    .dx-item {
        font-size: 14px;
    }

    .dx-field-item-label {
        padding: 0;
        font-family: Lato;
    }

    .dx-field-item-label-text {
        color: var(--themeLabelGrey);
        font-family: Lato;
        font-size: 12px;
        font-style: normal;
        font-weight: 700;
        line-height: 14.4px;
    }

    .disabled-field {
        .dx-field-item-label-text {
            color: var(--themeGreyBorder);
        }
    }

    .dx-texteditor-container {
        border-radius: 4px;
        display: inline-flex;
        align-items: center;
        gap: 10px;
        background-color: var(--white);
        border: 1px solid var(--themeSecondaryGreyColor);
        color: var(--themeSecondaryGreyColor);
        font-family: Lato;
        font-size: 14px;
        font-style: normal;
        font-weight: 400;
        line-height: normal;
    }

    .dx-texteditor.dx-editor-filled:after {
        border-bottom: none;
    }

    .dx-texteditor.dx-editor-filled .dx-placeholder:before,
    .dx-texteditor.dx-editor-filled .dx-texteditor-input {
        padding: 10px;
    }

    .dx-texteditor-input-container {
        width: 194px;
        height: 40px;
    }
    .custom-tag {
        .dx-texteditor-input-container {
            width: 194px;
            height: auto;
          }
    }
}