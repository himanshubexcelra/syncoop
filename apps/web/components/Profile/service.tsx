"use server";

export async function getUsersById(withRelation: string[] = [], id: number) {
    const url = new URL(`${process.env.API_HOST_URL}/v1/users`);
    if (withRelation.length) {
        url.searchParams.append('with', JSON.stringify(withRelation));
    }
    if (id) {
        url.searchParams.append('id', JSON.stringify(id));
    }
    const response = await fetch(url, {
        mode: "no-cors",
        method: "GET",
        headers: {
            "Content-Type": "application/json",
        },
    });
    const data = await response.json();
    console.log(data)
    return data;
}