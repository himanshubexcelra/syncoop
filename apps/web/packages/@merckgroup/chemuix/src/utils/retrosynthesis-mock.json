[
  {
    "aromatize_smiles": "CC(=O)NC1=CC=CC=C1Br.CCOC(=O)CCl>>CCOC(=O)CC1=CC=CC=C1NC(C)=O",
    "conditions": "[Ir]-photocat.[Ni]-cat.silane reagent.base.blue light",
    "inventory": "Not available",
    "name": "Photoredox Cross-Electrophile Coupling of alpha-Chloro Carbonyls with Aryl Halides",
    "product_inchi_key": "QXKDASDRIZQXDH-UHFFFAOYSA-N",
    "product_molecular_formula": "C12H15NO3",
    "product_molecular_weight": 221.256,
    "product_smile": "CCOC(=O)Cc1ccccc1NC(C)=O",
    "products": "CCOC(=O)Cc1ccccc1NC(C)=O",
    "protections": "Not available",
    "reactant_1_inchi_key": "VOBKUOHHOWQHFZ-UHFFFAOYSA-N",
    "reactant_1_molecular_formula": "C8H8BrNO",
    "reactant_1_molecular_weight": 214.06199999999995,
    "reactant_1_smile": "CC(=O)Nc1ccccc1Br",
    "reactant_2_inchi_key": "VEUUMBGHMNQHGO-UHFFFAOYSA-N",
    "reactant_2_molecular_formula": "C4H7ClO2",
    "reactant_2_molecular_weight": 122.551,
    "reactant_2_smile": "CCOC(=O)CCl",
    "reactants": ["CC(=O)Nc1ccccc1Br", "CCOC(=O)CCl"],
    "reaction_details": "CC(=O)Nc1ccccc1Br.CCOC(=O)CCl>>CCOC(=O)Cc1ccccc1NC(C)=O",
    "reference": "[10.1002/anie.201909072]: https://doi.org/10.1002/anie.201909072"
  },
  {
    "aromatize_smiles": "CCOC(=O)CC1=CC=CC=C1NC(C)=O.O=S(Cl)Cl>>CC(=O)NC1=CC=CC=C1CC(=O)Cl",
    "conditions": "1. LiOH.H2O.THF 2.SOCl2.DCM or DMF",
    "inventory": "Not available",
    "name": "Synthesis of acyl chlorides from esters",
    "product_inchi_key": "PUKCNUTYODCILT-UHFFFAOYSA-N",
    "product_molecular_formula": "C10H10ClNO2",
    "product_molecular_weight": 211.64799999999994,
    "product_smile": "CC(=O)Nc1ccccc1CC(=O)Cl",
    "products": "CC(=O)Nc1ccccc1CC(=O)Cl",
    "protections": "Not available",
    "reactant_1_inchi_key": "QXKDASDRIZQXDH-UHFFFAOYSA-N",
    "reactant_1_molecular_formula": "C12H15NO3",
    "reactant_1_molecular_weight": 221.256,
    "reactant_1_smile": "CCOC(=O)Cc1ccccc1NC(C)=O",
    "reactant_2_inchi_key": "FYSNRJHAOHDILO-UHFFFAOYSA-N",
    "reactant_2_molecular_formula": "Cl2OS",
    "reactant_2_molecular_weight": 118.97200000000001,
    "reactant_2_smile": "O=S(Cl)Cl",
    "reactants": ["CCOC(=O)Cc1ccccc1NC(C)=O", "O=S(Cl)Cl"],
    "reaction_details": "CCOC(=O)Cc1ccccc1NC(C)=O.O=S(Cl)Cl>>CC(=O)Nc1ccccc1CC(=O)Cl",
    "reference": "[10.1016/j.tet.2007.04.043]: https://doi.org/10.1016/j.tet.2007.04.043 [10.3390/50500714]: https://doi.org/10.3390/50500714 [10.1002/adsc.200303011]: https://doi.org/10.1002/adsc.200303011 [10.1021/ja073476s]: https://doi.org/10.1021/ja073476s"
  },
  {
    "aromatize_smiles": "C#CCC1CCCCC1.CC(=O)NC1=CC=CC=C1CC(=O)Cl>>CC(=O)NC1=CC=CC2=C(CC3CCCCC3)C=C(O)C=C12",
    "conditions": "AlCl3.DCM.rt",
    "inventory": "Not available",
    "name": "Synthesis of 2-naphtols from alkynes",
    "product_inchi_key": "VDBBCHLKIPFSRA-UHFFFAOYSA-N",
    "product_molecular_formula": "C19H23NO2",
    "product_molecular_weight": 297.3979999999999,
    "product_smile": "CC(=O)Nc1cccc2c(CC3CCCCC3)cc(O)cc12",
    "products": "CC(=O)Nc1cccc2c(CC3CCCCC3)cc(O)cc12",
    "protections": "Not available",
    "reactant_1_inchi_key": "UARFKZSJGDQRLF-UHFFFAOYSA-N",
    "reactant_1_molecular_formula": "C9H14",
    "reactant_1_molecular_weight": 122.21100000000001,
    "reactant_1_smile": "C#CCC1CCCCC1",
    "reactant_2_inchi_key": "PUKCNUTYODCILT-UHFFFAOYSA-N",
    "reactant_2_molecular_formula": "C10H10ClNO2",
    "reactant_2_molecular_weight": 211.64799999999994,
    "reactant_2_smile": "CC(=O)Nc1ccccc1CC(=O)Cl",
    "reactants": ["C#CCC1CCCCC1", "CC(=O)Nc1ccccc1CC(=O)Cl"],
    "reaction_details": "C#CCC1CCCCC1.CC(=O)Nc1ccccc1CC(=O)Cl>>CC(=O)Nc1cccc2c(CC3CCCCC3)cc(O)cc12",
    "reference": "[10.1021/ol502951v]: https://doi.org/10.1021/ol502951v"
  },
  {
    "aromatize_smiles": "CC(=O)NC1=CC=CC2=C(CC3CCCCC3)C=C(O)C=C12>>CC(=O)NC1=CC=CC2=C(C(=O)C3CCCCC3)C=C(O)C=C12",
    "conditions": "oxidant eg. Oxone or O2 or K2S2O8",
    "inventory": "Not available",
    "name": "Benzylic oxidation to ketone",
    "product_inchi_key": "WMLCTPFIJIXTOJ-UHFFFAOYSA-N",
    "product_molecular_formula": "C19H21NO3",
    "product_molecular_weight": 311.381,
    "product_smile": "CC(=O)Nc1cccc2c(C(=O)C3CCCCC3)cc(O)cc12",
    "products": "CC(=O)Nc1cccc2c(C(=O)C3CCCCC3)cc(O)cc12",
    "protections": "Not available",
    "reactant_1_inchi_key": "VDBBCHLKIPFSRA-UHFFFAOYSA-N",
    "reactant_1_molecular_formula": "C19H23NO2",
    "reactant_1_molecular_weight": 297.3979999999999,
    "reactant_1_smile": "CC(=O)Nc1cccc2c(CC3CCCCC3)cc(O)cc12",
    "reactants": ["CC(=O)Nc1cccc2c(CC3CCCCC3)cc(O)cc12"],
    "reaction_details": "CC(=O)Nc1cccc2c(CC3CCCCC3)cc(O)cc12>>CC(=O)Nc1cccc2c(C(=O)C3CCCCC3)cc(O)cc12",
    "reference": "[10.1021/acs.orglett.6b02914]: https://doi.org/10.1021/acs.orglett.6b02914 [10.1021/ol0520020]: https://doi.org/10.1021/ol0520020 [10.1055/s-0037-1610678]: https://doi.org/10.1055/s-0037-1610678 [10.1021/ol300853z]: https://doi.org/10.1021/ol300853z"
  }
]
