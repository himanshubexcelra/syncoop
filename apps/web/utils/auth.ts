'use server'

import { cookies } from "next/headers";
import { Messages } from "./message";

export async function authorize(formData: FormData) {
    try {
        const body = JSON.stringify({
            email: formData.get('email'),
            password: formData.get('password')
        });
        const response: any = await fetch(`${process.env.API_HOST_URL}/v1/auth`, {
            mode: 'no-cors',
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body
        });
        console.log(response);
        if (response.status === 200) {
            const output = await response.json();
            const sessionData = JSON.stringify(output.data);
            const cookie = cookies();
            cookie.set('session', sessionData, {
                httpOnly: true,
                secure: process.env.NODE_ENV === 'production',
                maxAge: 60 * 60 * 24 * 7, // One week
                path: '/',
            });
            return output;
        } else if (response.status === 404) {
            const output = await response.json();
            return output;
        }
        throw new Error(Messages.SOMETHING_WENT_WRONG);
    }
    catch (error) {
        return error;
    }
}

export async function clearSession() {
    cookies().delete('session');
}

export async function getUserData() {
    const sessionData: any = cookies().get('session')?.value;
    if (sessionData) {
        return JSON.parse(sessionData);
    }
    return null;
}