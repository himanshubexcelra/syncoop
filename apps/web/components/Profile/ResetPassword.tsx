/*eslint max-len: ["error", { "code": 100 }]*/
"use client";
import toast from "react-hot-toast";
import Image from 'next/image';
import { Button, Tooltip } from 'devextreme-react';
import { LoginFormSchema } from '@/lib/definition';
import { Messages } from '@/utils/message';
import PasswordCriteria from "../Tooltips/PasswordCriteria";
import { useCallback, useRef, useState } from 'react';
import { delay, generatePassword } from "@/utils/helpers";
import {
    ButtonItem,
    ButtonOptions,
    Form, GroupItem, Item, Label, SimpleItem
} from "devextreme-react/form";
import { editUser } from "../User/service";
import { DELAY } from "@/utils/constants";

interface ChangePasswordProps {
    onClose: () => void;
    email_id?: string;
}
const customPasswordCheck = (password: any) =>
    LoginFormSchema.shape.password_hash.safeParse(password).success;


export default function ResetPassword({ onClose, email_id,
}: ChangePasswordProps) {
    const formRef = useRef<any>(null);
    const [newPassword, setNewPassword] = useState('');

    const handleSubmit = async () => {
        const values = formRef.current!.instance().option("formData");
        const validationCheck = customPasswordCheck(values.newPassword)
        if (validationCheck) {
            const response = await editUser({ ...values, email_id });
            if (response.status === 200) {
                setNewPassword('');
                formRef.current!.instance().reset();
                onClose();
                const toastId = toast.success(Messages.PASSWORD_CHANGE)
                await delay(DELAY)
                toast.remove(toastId)
            } else {
                const toastId = toast.error(`${response.error}`);
                await delay(DELAY);
                toast.remove(toastId);
            }
        }
        else {
            formRef.current!.instance().validate()
        }
    };
    let copyPassword = newPassword
    const handleCopyPassword = async () => {
        if (copyPassword === "") {
            const toastId = toast.error(Messages.PASSWORD_EMPTY);
            await delay(DELAY);
            toast.remove(toastId);
            return;
        }
        navigator.clipboard.writeText(copyPassword)
            .then(() => toast.success(Messages.PASSWORD_COPY))
            .catch(() => toast.error(Messages.PASSWORD_COPY_FAIL));
    };

    const handleGeneratePassword = () => {
        const generatedPassword = generatePassword();
        setNewPassword(generatedPassword);
        navigator.clipboard.writeText(generatedPassword)
            .then(() => toast.success(Messages.PASSWORD_COPY))
            .catch(() => toast.error(Messages.PASSWORD_COPY_FAIL));
        formRef.current!.instance().updateData("newPassword", generatedPassword);
    };


    const handleCancel = () => {
        setNewPassword('');
        formRef.current!.instance().reset();
        onClose();
    }
    const changePasswordMode = useCallback((name: any) => {
        const editor = formRef.current.instance().getEditor(name);
        const currentMode = editor.option('mode');
        editor.option('mode', currentMode === 'text' ? 'password' : 'text');
        const passwordButton = editor.option('buttons').find((button: any) =>
            button.name === name);
        if (passwordButton) {
            passwordButton.options.icon = currentMode === 'text' ? "eyeopen" : "eyeclose";
        }
        editor.repaint();
    }, []);
    return (
        <>
            <div className="flex justify-between mb-4">
                <div className="subHeading">Generate Password</div>
                <Image
                    className='cursor-pointer'
                    src="/icons/cross-icon.svg"
                    alt="close icon"
                    width={21.1}
                    height={22.5}
                    onClick={handleCancel}
                />
            </div>
            <Form ref={formRef}>
                {/* <SimpleItem
                    dataField="newPassword"
                    render={newPasswordRender}
                /> */}
                <GroupItem colCount={4} cssClass="reset-password-group">
                    <SimpleItem
                        dataField="newPassword"
                        editorType="dxTextBox"
                        cssClass="custom-password"
                        editorOptions={{
                            mode: 'password',
                            placeholder: "Enter Password",
                            valueChangeEvent: 'keyup',
                            onValueChanged: (e: any) => copyPassword = e.value,
                            buttons: [{
                                name: "newPassword",
                                location: "after",
                                // options: passwordButton,
                                options: {
                                    stylingMode: 'text',
                                    icon: 'eyeopen',
                                    onClick: () => changePasswordMode('newPassword'),
                                },
                            }],
                        }}
                        validationRules={[
                            {
                                type: 'required',
                                message: Messages.requiredMessage('Password')
                            },
                            {
                                type: 'custom',
                                validationCallback: (e) => {
                                    return customPasswordCheck(e.value);
                                },
                                message: Messages.PASSWORD_CRITERIA
                            }
                        ]}
                    >
                        <Label text={'Password'} />
                    </SimpleItem>
                    <Item>
                        <div className="mt-[12px]">
                            <Image
                                src="/icons/copy-icon.svg"
                                alt="copy"
                                width={16}
                                height={16}
                                priority
                                onClick={handleCopyPassword}
                            />
                        </div>
                    </Item>
                    <Item>
                        <div className="mt-[12px]" id="info-box">
                            <Image
                                src="/icons/info-icon.svg"
                                alt="info"
                                width={16}
                                height={16}
                                priority
                                id="info-icon-password_hash"
                            />
                            <Tooltip
                                target="#info-icon-password_hash"
                                showEvent="mouseenter"
                                hideEvent="mouseleave"
                                position="bottom"
                                hideOnOutsideClick={false}
                            >
                                <PasswordCriteria />
                            </Tooltip>
                        </div>
                    </Item>
                    <ButtonItem cssClass="pt-[20px]">
                        <ButtonOptions
                            text={`Generate`}
                            onClick={handleGeneratePassword}
                            elementAttr={{ class: 'btn-secondary' }} />
                    </ButtonItem>
                </GroupItem>
                <SimpleItem>
                    <div className="flex justify-start gap-2 mt-5 ">
                        <Button
                            text="Update"
                            onClick={handleSubmit}
                            useSubmitBehavior={true}
                            hoverStateEnabled={false}
                            elementAttr={{ class: "btn-primary" }}
                        />
                        <Button
                            text="Cancel"
                            onClick={handleCancel}
                            elementAttr={{ class: "btn-secondary" }}
                        />
                    </div>
                </SimpleItem>
            </Form>
        </>
    );
}
