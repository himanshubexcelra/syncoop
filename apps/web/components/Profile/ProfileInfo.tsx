"use client";
import { useRef, useState, useEffect } from 'react';
import styles from './ProfileInfo.module.css'
import { HeadingObj, OrgUser, UserData, } from '@/lib/definition';
import { Popup as MainPopup, } from "devextreme-react/popup";
import { Button } from "devextreme-react/button";
import { getUsersById } from './service';
import DialogPopUp from '@/ui/DialogPopUp';
import ChangePassword from './ChangePassword';
import ResetPassword from './ResetPassword';
import Heading from '../Heading/Heading';
import RenderEditUser from '../User/editUserDetails';
import { Messages } from '@/utils/message';
import Breadcrumb from '../Breadcrumbs/BreadCrumbs';
import { getProfileBreadCrumbs } from './breadCrumbs';

const changeDialogProperties = {
    width: 480,
    height: 338,
}
const resetDialogProperties = {
    width: 480,
    height: 260,
}

type ProfileInfoProps = {
    id: number;
    myRoles: string[];
    isMyProfile: boolean;
    actionsEnabled: string[]
    orgDetailLoggedIn?: OrgUser
}

export default function ProfileInfo({ id,
    myRoles,
    isMyProfile,
    actionsEnabled,
    orgDetailLoggedIn
}: ProfileInfoProps) {
    const [data, setData] = useState<UserData[]>([]);
    const [formVisible, setFormVisible] = useState(false);
    const [passwordPopupVisible, setPasswordPopupVisible] = useState(false);
    const formRef = useRef<any>(null);
    const { email_id, first_name, last_name, orgUser, user_role, status } = data[0] || {}
    const userRoleNames = user_role?.map(role => role.role.name).join(', ');

    const formEdit = {
        email_id, first_name, user_role, last_name, orgUser
    }

    const hidePopup = () => {
        setPasswordPopupVisible(false)
    };

    const contentProps = {
        email_id,
        onClose: hidePopup
    }

    const fetchData = async () => {
        try {
            const userInfo = await getUsersById(['orgUser', 'user_role'], id);
            setData(userInfo)

        } catch (error) {
            console.error(Messages.USER_FETCH_ERROR, error);
        }
    };

    useEffect(() => {
        fetchData();
    }, [id]);

    const heading: HeadingObj[] = [
        {
            svgPath: "/icons/profile-icon-lg-active.svg",
            label: first_name ? `${first_name}` : '',
            svgWidth: 33,
            svgHeight: 36,
            href: "",
            type: "Profile:"
        },
    ];
    const breadcrumbs = getProfileBreadCrumbs(isMyProfile, myRoles, orgDetailLoggedIn || null)
    return (
        <>
            <Breadcrumb breadcrumbs={breadcrumbs} />
            <Heading heading={heading} myRoles={myRoles} />
            <main className={styles.main}>
                <div className={styles.box}>
                    <div className="flex justify-between items-center mb-4">
                        <div className="groupItem">My Profile</div>
                        {(isMyProfile || actionsEnabled.includes('edit_user') || myRoles.includes('admin')) &&
                            <div className="space-x-2">
                                <Button className="btn-primary" onClick={() => setPasswordPopupVisible(true)}>
                                    {isMyProfile ? "Change Password" : "Reset Password"}
                                </Button>
                                <Button className="btn-secondary" onClick={() => setFormVisible(true)}>
                                    Edit
                                </Button>
                            </div>}
                    </div>
                    <div className={`${styles.grid} gap-y-5`}>
                        <div>
                            <span className={`subHeading ${styles.type}`}>Organization:</span>
                            <span className={`subHeading ${styles.output}`}> {orgUser?.name}</span>
                        </div>
                        <div>
                            <span className={`subHeading ${styles.type}`}>Roles:</span>
                            <span className={`subHeading ${styles.output}`}> {userRoleNames}</span>
                        </div>
                        <div>
                            <span className={`subHeading ${styles.type}`}>First Name:</span>
                            <span className={`subHeading ${styles.output}`}> {first_name}</span>
                        </div>
                        <div>
                            <span className={`subHeading ${styles.type}`}>email_id:</span>
                            <span className={`subHeading ${styles.output}`}> {email_id}</span>
                        </div>
                        <div>
                            <span className={`subHeading ${styles.type}`}>Last Name:</span>
                            <span className={`subHeading ${styles.output}`}> {last_name}</span>
                        </div>
                        {!isMyProfile && <div>
                            <span className={`subHeading ${styles.type}`}>Status:</span>
                            <span className={`subHeading ${styles.output}`}> {status}</span>
                        </div>}
                    </div>
                </div>
                <MainPopup
                    title="Edit Profile"
                    visible={formVisible}
                    contentRender={() => (
                        <RenderEditUser
                            tableData={formEdit}
                            formRef={formRef}
                            setCreatePopupVisibility={setFormVisible}
                            roles={[]}
                            isMyProfile={isMyProfile}
                            fetchData={fetchData}
                            myRoles={myRoles} />
                    )}
                    width={470}
                    hideOnOutsideClick={true}
                    height="100%"
                    position={{
                        my: { x: 'right', y: 'top' },
                        at: { x: 'right', y: 'top' },
                    }}
                    onHiding={() => {
                        setFormVisible(false);
                        formRef.current!.instance().reset();
                    }}
                    showCloseButton={true}
                    wrapperAttr={{ class: "create-popup mr-[15px]" }} />
                <DialogPopUp {...{
                    visible: passwordPopupVisible,
                    dialogProperties: isMyProfile ? changeDialogProperties : resetDialogProperties,
                    Content: isMyProfile ? ChangePassword : ResetPassword,
                    hidePopup,
                    contentProps
                }} /></main></>
    )
}