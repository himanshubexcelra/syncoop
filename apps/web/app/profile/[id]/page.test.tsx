import { render, screen } from "@testing-library/react";
import Profile from "./page";
import { getUserData } from "@/utils/auth";
import ProfileInfo from "@/components/Profile/ProfileInfo";

jest.mock("@/components/Profile/ProfileInfo", () => ({
    __esModule: true,
    default: jest.fn(() => <div data-testid="profile-info">Mocked ProfileInfo</div>),
}));


jest.mock('@/utils/auth', () => ({
    getUserData: jest.fn(),
}));


jest.mock("@/components/layout", () => ({
    __esModule: true,
    default: jest.fn(({ children }) => <div data-testid="layout">{children}</div>),
}));

describe("Profile Page", () => {
    const mockSessionData = {
        userData: {
            orgUser: { id: 1, name: "FauxBio" },
            myRoles: ['admin']
        },
        actionsEnabled: ['edit_user'],
        isMyProfile: true,
    };

    const params = { id: "2" };

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it.skip("should render layout and profile info when user data is available", async () => {
        (getUserData as jest.Mock).mockResolvedValue(mockSessionData);

        expect(screen.getByTestId("layout")).toBeInTheDocument();
        expect(screen.getByTestId("profile-info")).toBeInTheDocument();
    });

    it("should pass correct props to ProfileInfo component", async () => {
        (getUserData as jest.Mock).mockResolvedValue(mockSessionData);

        render(await Profile({ params }));

        expect(ProfileInfo).toHaveBeenCalledWith({
            id: 2,
            myRoles: mockSessionData.userData.myRoles,
            isMyProfile: false,
            actionsEnabled: mockSessionData.actionsEnabled,
            orgDetailLoggedIn: mockSessionData.userData.orgUser
        }, {});
    });

});