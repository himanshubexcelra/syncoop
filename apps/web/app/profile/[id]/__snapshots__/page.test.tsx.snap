// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page should render layout and profile info when user data is available 1`] = `
<div
  data-testid="layout"
>
  <div
    data-testid="profile-info"
  >
    Mocked ProfileInfo
  </div>
</div>
`;
